pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -213,18 +214,21 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
 
     @Override
     public ProducerBuilder<T> batchingMaxPublishDelay(long batchDelay, @NonNull TimeUnit timeUnit) {
+        checkArgument(batchDelay >= 0, "batchDelay needs to be >= 0");
         conf.setBatchingMaxPublishDelayMicros(timeUnit.toMicros(batchDelay));
         return this;
     }
 
     @Override
     public ProducerBuilder<T> batchingMaxMessages(int batchMessagesMaxMessagesPerBatch) {
+        checkArgument(batchMessagesMaxMessagesPerBatch >= 0, "batchMessagesMaxMessagesPerBatch needs to be >= 0");
```

Comment:
Why this can’t be negative if batching is disabled?

Reply:
There is the similar check on following flows:
- _deprecated_ `ProducerConfiguration.setBatchingMaxMessages` 
- `ProducerConfiguration.cc` => `setBatchingMaxMessages`

Reply:
if ProducerConfiguration already verifies it, why do we need to verify it in the producer builder? The implementation of builders are wrappers for configuration objects.

Also ProducerConfiguration.setBatchingMaxMessages is verifying `> 0`. but you are adding one for `>= 0`. This is a problem, no?

Reply:
Same comment as above.

Also, there is following mismatch on **legacy C++ and Java Client APIs**. Alignment can also be useful here if makes sense.

`
ProducerConfiguration.cc ## setBatchingMaxMessages => "batchingMaxMessages needs to be greater than 1"
ProducerConfiguration.setBatchingMaxMessages => checkArgument(batchMessagesMaxMessagesPerBatch > 0);
`

Reply:
Same comments as above.

Reply:
Sure, will be addressing this as well.